client2/app/components/AddAssignmentForm.tsx

```diff
@@ -0,0 +1,68 @@
+"use client";
+import React, { ReactNode, useState } from "react";
+
+export const AddAssignmentForm = ({ accountId }: { accountId: string }):ReactNode => {
+  const [title, setTitle] = useState("");
+  const [score, setScore] = useState<number>(0);
+  const [description, setDescription] = useState("");
+  const [category, setCategory] = useState("");
+
+  const handleTitleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+  e.preventDefault();
+  setTitle(e.target.value)
+  }
+
+    const handleScoreChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    e.preventDefault();
+    setScore(Number(e.target.value))
+    }
+
+    const handleDescriptionChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+      e.preventDefault();
+      setDescription(e.target.value)
+      }
+
+      const handleSelectCategory = (e: React.ChangeEvent<HTMLSelectElement>) => {
+        e.preventDefault();
+        setCategory(e.target.value)
+        console.log(category)
+        }
+
+  const submitAssignment = async (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    const payload = JSON.stringify({
```

Comment:
Isn’t it better to interface or type the object instead of adding manual fields?